src/react-router.tsx

```diff
@@ -14,13 +14,15 @@ const preservedRoutes = generatePreservedRoutes<Element>(PRESERVED)
 
 const regularRoutes = generateRegularRoutes<RouteObject, () => Promise<Module>>(ROUTES, (module, key) => {
   const Element = lazy(module)
+  const ErrorBoundary = lazy(() => module().then((module) => ({ default: module.ErrorBoundary })))
```

Comment:
What about having a similar name as we had initially w/ React Location? we used `Pending` for loading and `Failure` for errors. If you could rename `ErrorBoundary` to `Failure` in this case it would be great, to keep consistent with the other integrations at the moment.

Here's a quick suggestion:

```suggestion
  const Failure = lazy(() => module().then((mod) => ({ default: mod.Failure })))
```

Reply:
PS: `Failure` is inspired by Redwood.js https://redwoodjs.com/docs/cells#failure. `Pending` was initially `Loading` but renamed to be less confusing after the `Loader` function addition if I recall correctly.

Reply:
Good point, however, Failure does not sound good I guess. I have a feeling being closer to Remix/React-Router would be the way to go as Next.JS are basically copying most of the concepts so they are becoming the norm. 

Failure may be confusing as I would expect my error pages to be my Failures. The ErrorBoundary inside the same file just makes more sense to me.

Reply:
That makes sense to me as well, but I'm leaning more towards `ErrorElement` to exactly match the prop React Router and React Location exposes at route object for the error element. What do you think?

Reply:
It just does not sound right, let's just call it Error?
This way is's consistent with the other primitives.

Reply:
That was also my first thought as well, but I'm afraid it can conflict with the `Error` constructor used to throw errors `throw new Error('...')`

Reply:
I tried and I can confirm that it would conflict:

![Screenshot_20221113-233847~2](https://user-images.githubusercontent.com/20151138/201548320-28e00ac0-1e86-4cc2-88c3-8aa235e3989e.png)